### PR TITLE
GCS: update RC override to obey mavlink spec at GCS handler

### PR DESCRIPTION
**Pull request by magicrub**
update RC override to obey mavlink spec at GCS handler. We were technically still rejecting UINT16_MAX values but they were checked down in the RC_Channel class. This adds a check up in the mavlink handler where it belongs so it's obvious that we're following the spec.

### CODE REVIEW

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -3009,22 +3009,31 @@ void GCS_MAVLINK::handle_rc_channels_override(const mavlink_message_t *msg)
...
+            packet.chan10_raw,
+            packet.chan11_raw,
+            packet.chan12_raw,
+            packet.chan13_raw,
+            packet.chan14_raw,
+            packet.chan15_raw,
+            packet.chan16_raw
+    };
+
+    for (uint8_t i=0; i<16; i++) {
```
Comment (peterbarker) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: Should be `ARRAY_SIZE`.

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 2]
```diff
@@ -3009,22 +3009,31 @@ void GCS_MAVLINK::handle_rc_channels_override(const mavlink_message_t *msg)
...
-    RC_Channels::set_override(7, packet.chan8_raw, tnow);
-    RC_Channels::set_override(8, packet.chan9_raw, tnow);
-    RC_Channels::set_override(9, packet.chan10_raw, tnow);
-    RC_Channels::set_override(10, packet.chan11_raw, tnow);
-    RC_Channels::set_override(11, packet.chan12_raw, tnow);
-    RC_Channels::set_override(12, packet.chan13_raw, tnow);
-    RC_Channels::set_override(13, packet.chan14_raw, tnow);
-    RC_Channels::set_override(14, packet.chan15_raw, tnow);
-    RC_Channels::set_override(15, packet.chan16_raw, tnow);
+    const uint16_t override_data[16] = {
```
Comment (peterbarker) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: Can this be `[]`?

File: libraries/RC_Channel/RC_Channel.cpp [hunk 3]
```diff
@@ -332,9 +332,8 @@ void RC_Channel::set_override(const uint16_t v, const uint32_t timestamp_us)
     if (!rc().gcs_overrides_enabled()) {
         return;
     }
-    // this UINT16_MAX stuff should really, really be in the
-    // mavlink packet handling code.  It can be moved once that
-    // code is in the GCS_MAVLink class!
+
+    // this UINT16_MAX check is also done in GCS_MAVLink handling to obey the MAVLink spec
     if (v == UINT16_MAX) {
```
Comment (peterbarker) [inline libraries/RC_Channel/RC_Channel.cpp]: Why keep this?

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: Also, what does this do to the fmuv3 binary size?

File: libraries/RC_Channel/RC_Channel.cpp (hunk 3 above)
Comment (magicrub) [reply libraries/RC_Channel/RC_Channel.cpp]: gone

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 2 above)
Comment (magicrub) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: changed to [ ] and program size delta:

|Target     |          Text |   Data |BSS    | Total|
| - | - | - | - | - |
|bin/ardurover  | -48   |      0  |   0   |     -48|

So, it's smaller.

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 1 above)
Comment (magicrub) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: fixed

**Comment by magicrub** [comment]
@peterbarker fixed everything, and flash size goes down by 48 bytes.

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 4]
```diff
@@ -3009,22 +3009,31 @@ void GCS_MAVLINK::handle_rc_channels_override(const mavlink_message_t *msg)
...
+            packet.chan10_raw,
+            packet.chan11_raw,
+            packet.chan12_raw,
+            packet.chan13_raw,
+            packet.chan14_raw,
+            packet.chan15_raw,
+            packet.chan16_raw
+    };
+
+    for (uint8_t i=0; i<ARRAY_SIZE(override_data); i++) {
```
Comment (OXINARF) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: @magicrub I'll pay whatever money to fix your space bar 😄 One day I'll understand why so many devs are afraid of spaces 😛

Comment (magicrub) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: should just be one tab. I'll fix that

**Comment by magicrub** [comment]
whoops, I thought I added that email to github. Fixed now. Thanks for the reminder!

**Comment by magicrub** [comment]
Thanks for the review @OXINARF and @peterbarker !

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 4 above)
Comment (OXINARF) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: I was referring to spaces surrounding operators (=, <) 😜

**Comment by peterbarker** [comment]
Nice one.  Always good to see a FIXME go away :-)

**Pull request merged**